File: frontend/assets/icons/gen_icons.js
const fs = require('fs');
const zlib = require('zlib');
const path = require('path');

const SIZE = 81;
const OUT = path.resolve(__dirname);

const GRAY   = [153, 153, 153, 255];
const ORANGE = [201, 126,  90, 255];
const CLEAR  = [0, 0, 0, 0];

function pngChunk(type, data) {
  const len = Buffer.allocUnsafe(4);
  len.writeUInt32BE(data.length, 0);
  const t = Buffer.from(type, 'ascii');
  const crc = Buffer.allocUnsafe(4);
  crc.writeUInt32BE(zlib.crc32(Buffer.concat([t, data])), 0);
  return Buffer.concat([len, t, data, crc]);
}

function makePNG(pixels) {
  // pixels: SIZE x SIZE array of [r,g,b,a]
  const sig = Buffer.from([0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A]);
  const ihdrData = Buffer.allocUnsafe(13);
  ihdrData.writeUInt32BE(SIZE, 0);
  ihdrData.writeUInt32BE(SIZE, 4);
  ihdrData.writeUInt8(8, 8);   // bit depth
  ihdrData.writeUInt8(6, 9);   // RGBA
  ihdrData.writeUInt8(0, 10);  // compression
  ihdrData.writeUInt8(0, 11);  // filter
  ihdrData.writeUInt8(0, 12);  // interlace

  const raw = Buffer.allocUnsafe(SIZE * (1 + SIZE * 4));
  let p = 0;
  for (let y = 0; y < SIZE; y++) {
    raw[p++] = 0; // filter: None
    for (let x = 0; x < SIZE; x++) {
      const c = pixels[y][x];
      raw[p++] = c[0];
      raw[p++] = c[1];
      raw[p++] = c[2];
      raw[p++] = c[3];
    }
  }

  const idatData = zlib.deflateSync(raw, { level: 9 });
  return Buffer.concat([
    sig,
    pngChunk('IHDR', ihdrData),
    pngChunk('IDAT', idatData),
    pngChunk('IEND', Buffer.alloc(0))
  ]);
}

function canvas() {
  return Array.from({ length: SIZE }, () =>
    Array.from({ length: SIZE }, () => [...CLEAR])
  );
}

function setPx(c, x, y, col) {
  if (x >= 0 && x < SIZE && y >= 0 && y < SIZE) c[y][x] = [...col];
}

function fillRect(c, x1, y1, x2, y2, col) {
  for (let y = Math.max(0, y1); y <= Math.min(SIZE - 1, y2); y++) {
    for (let x = Math.max(0, x1); x <= Math.min(SIZE - 1, x2); x++) {
      c[y][x] = [...col];
    }
  }
}

function drawCircle(c, cx, cy, r, col) {
  for (let y = Math.max(0, cy - r); y <= Math.min(SIZE - 1, cy + r); y++) {
    for (let x = Math.max(0, cx - r); x <= Math.min(SIZE - 1, cx + r); x++) {
      const d = Math.sqrt((x - cx) ** 2 + (y - cy) ** 2);
      if (d <= r) c[y][x] = [...col];
    }
  }
}

// ─── 图标绘制 ──────────────────────────────────────────

function iconHome(col) {
  const c = canvas();
  const m = SIZE >> 1;
  // 屋顶三角
  for (let i = 0; i < 28; i++) {
    fillRect(c, m - i - 2, 18 + i, m - i + 2, 18 + i + 2, col); // 左斜
    fillRect(c, m + i - 2, 18 + i, m + i + 2, 18 + i + 2, col); // 右斜
  }
  // 房身
  fillRect(c, 22, 46, 59, 64, col);
  // 门洞
  fillRect(c, 33, 50, 48, 64, CLEAR);
  fillRect(c, 35, 50, 46, 64, col);
  return c;
}

function iconReserve(col) {
  const c = canvas();
  // 外框
  fillRect(c, 14, 18, 67, 65, col);
  fillRect(c, 18, 22, 63, 61, CLEAR);
  // 顶部条
  fillRect(c, 14, 18, 67, 30, col);
  // 挂钉
  fillRect(c, 24, 12, 30, 22, col);
  fillRect(c, 51, 12, 57, 22, col);
  // 网格点
  for (let row = 0; row < 3; row++) {
    for (let j = 0; j < 3; j++) {
      const x = 22 + j * 15;
      const y = 36 + row * 10;
      fillRect(c, x, y, x + 5, y + 5, col);
    }
  }
  return c;
}

function iconMenu(col) {
  const c = canvas();
  // 碗（下半圆）
  const cx = 40, cy = 52, r = 22;
  for (let y = Math.max(0, cy - r); y <= Math.min(SIZE - 1, cy + r); y++) {
    for (let x = Math.max(0, cx - r); x <= Math.min(SIZE - 1, cx + r); x++) {
      const d = Math.sqrt((x - cx) ** 2 + (y - cy) ** 2);
      if (d <= r && y >= cy - 4) {
        if (d >= r - 4) c[y][x] = [...col];
      }
    }
  }
  // 碗口
  fillRect(c, 18, 26, 62, 30, col);
  // 筷子
  for (let i = 0; i < 3; i++) {
    fillRect(c, 26 + i * 8, 8, 28 + i * 8, 24, col);
  }
  return c;
}

function iconProfile(col) {
  const c = canvas();
  // 头
  drawCircle(c, 40, 25, 13, col);
  // 身体梯形
  for (let y = 42; y < 66; y++) {
    const w = 10 + ((y - 42) >> 1);
    for (let x = 40 - w; x <= 40 + w; x++) {
      setPx(c, x, y, col);
    }
  }
  return c;
}

const icons = {
  home:    iconHome,
  reserve: iconReserve,
  menu:    iconMenu,
  profile: iconProfile,
};

for (const [name, fn] of Object.entries(icons)) {
  // 普通态（灰色）
  fs.writeFileSync(path.join(OUT, `${name}.png`), makePNG(fn(GRAY)));
  console.log(`✅ ${name}.png`);
  // 激活态（橙色）
  fs.writeFileSync(path.join(OUT, `${name}_active.png`), makePNG(fn(ORANGE)));
  console.log(`✅ ${name}_active.png`);
}

console.log('\n🐱 全部 TabBar 图标生成完毕！');

File: frontend/assets/icons/gen_icons.py
"""
生成 NekoCafé TabBar 图标（极简线条风格 PNG）
普通态：灰色 #999999
激活态：暖橙 #C97E5A
"""
import struct, zlib, os

OUTPUT_DIR = r"D:\tmp\NekoCafe_Smart Reservation Platform\assets\icons"
SIZE = 81   # 微信 TabBar 建议 81×81

# ── 颜色 ────────────────────────────────────────────
GRAY   = (153, 153, 153, 255)   # #999
ORANGE = (201, 126,  90, 255)   # #C97E5A
TRANSPARENT = (0, 0, 0, 0)

def make_png(pixels):
    """将 SIZE×SIZE 的 RGBA 像素列表写成 PNG bytes"""
    def chunk(name, data):
        c = struct.pack(">I", len(data)) + name + data
        return c + struct.pack(">I", zlib.crc32(c[4:]) & 0xFFFFFFFF)

    raw = b""
    for row in pixels:
        raw += b"\x00"   # filter type None
        for px in row:
            raw += bytes(px)

    ihdr = struct.pack(">IIBBBBB", SIZE, SIZE, 8, 6, 0, 0, 0)
    idat = zlib.compress(raw, 9)

    return (
        b"\x89PNG\r\n\x1a\n"
        + chunk(b"IHDR", ihdr)
        + chunk(b"IDAT", idat)
        + chunk(b"IEND", b"")
    )

def empty_canvas():
    return [[list(TRANSPARENT) for _ in range(SIZE)] for _ in range(SIZE)]

def set_px(canvas, x, y, color):
    if 0 <= x < SIZE and 0 <= y < SIZE:
        canvas[y][x] = list(color)

def fill_rect(canvas, x1, y1, x2, y2, color):
    for y in range(y1, y2+1):
        for x in range(x1, x2+1):
            set_px(canvas, x, y, color)

def draw_circle(canvas, cx, cy, r, color, thickness=3):
    """绘制空心圆"""
    for y in range(cy-r-1, cy+r+2):
        for x in range(cx-r-1, cx+r+2):
            d = ((x-cx)**2 + (y-cy)**2)**0.5
            if r - thickness <= d <= r:
                set_px(canvas, x, y, color)

def draw_arc(canvas, cx, cy, r, start_deg, end_deg, color, thickness=3):
    import math
    for deg in range(int(start_deg), int(end_deg)):
        rad = math.radians(deg)
        for dr in range(-thickness//2, thickness//2+1):
            rx = int(round(cx + (r+dr) * math.cos(rad)))
            ry = int(round(cy + (r+dr) * math.sin(rad)))
            set_px(canvas, rx, ry, color)

def save(canvas, path):
    pixels = [[(c[0], c[1], c[2], c[3]) for c in row] for row in canvas]
    with open(path, "wb") as f:
        f.write(make_png(pixels))

# ════════════════════════════════════════════════════════
# 1. 首页图标（小屋形状）
# ════════════════════════════════════════════════════════
def icon_home(color):
    c = empty_canvas()
    # 屋顶三角（斜线）
    mid = SIZE // 2
    for i in range(30):
        # 左斜
        fill_rect(c, mid-i-3, 18+i, mid-i+3, 18+i, color)
        # 右斜
        fill_rect(c, mid+i-3, 18+i, mid+i+3, 18+i, color)
    # 房身矩形
    fill_rect(c, 22, 47, 59, 62, color)
    # 门
    fill_rect(c, 33, 50, 48, 62, TRANSPARENT)
    fill_rect(c, 35, 52, 46, 62, color)
    return c

# ════════════════════════════════════════════════════════
# 2. 预约图标（日历）
# ════════════════════════════════════════════════════════
def icon_reserve(color):
    c = empty_canvas()
    T = 3
    # 外框
    fill_rect(c, 14, 18, 67, 65, color)
    fill_rect(c, 14+T, 18+T, 67-T, 65-T, TRANSPARENT)
    # 顶部栏
    fill_rect(c, 14, 18, 67, 28, color)
    # 钉子
    fill_rect(c, 25, 13, 29, 23, color)
    fill_rect(c, 52, 13, 56, 23, color)
    # 网格点（3×3）
    for row in range(3):
        for col in range(3):
            x = 22 + col * 16
            y = 35 + row * 10
            fill_rect(c, x, y, x+5, y+5, color)
    return c

# ════════════════════════════════════════════════════════
# 3. 点单图标（碗+筷子）
# ════════════════════════════════════════════════════════
def icon_menu(color):
    c = empty_canvas()
    # 碗（下半圆）
    cx, cy, r = 40, 50, 24
    for y in range(cy-r, cy+r+1):
        for x in range(cx-r, cx+r+1):
            d = ((x-cx)**2 + (y-cy)**2)**0.5
            if d <= r and y >= cy-2:
                if d >= r-4:
                    set_px(c, x, y, color)
    # 碗口横线
    fill_rect(c, 16, 27, 64, 31, color)
    # 筷子
    for i in range(3):
        fill_rect(c, 27+i*8, 10, 29+i*8, 26, color)
    return c

# ════════════════════════════════════════════════════════
# 4. 我的图标（人形）
# ════════════════════════════════════════════════════════
def icon_profile(color):
    c = empty_canvas()
    # 头部圆
    draw_circle(c, 40, 24, 12, color, thickness=4)
    fill_rect(c, 30, 14, 50, 34, color)
    fill_rect(c, 32, 16, 48, 32, TRANSPARENT)
    # 实心头
    for y in range(13, 36):
        for x in range(29, 52):
            d = ((x-40)**2 + (y-24)**2)**0.5
            if d <= 12:
                set_px(c, x, y, color)
    # 身体（梯形）
    for y in range(42, 66):
        width = 12 + (y - 42) // 2
        x1 = 40 - width
        x2 = 40 + width
        for x in range(x1, x2+1):
            set_px(c, x, y, color)
    return c

# ════════════════════════════════════════════════════════
# 生成 8 个文件
# ════════════════════════════════════════════════════════
icons = {
    "home":    icon_home,
    "reserve": icon_reserve,
    "menu":    icon_menu,
    "profile": icon_profile,
}

for name, fn in icons.items():
    # 普通态（灰色）
    normal_path = os.path.join(OUTPUT_DIR, f"{name}.png")
    save(fn(GRAY), normal_path)
    print(f"✅ {name}.png")

    # 激活态（橙色）
    active_path = os.path.join(OUTPUT_DIR, f"{name}_active.png")
    save(fn(ORANGE), active_path)
    print(f"✅ {name}_active.png")

print("\n全部图标生成完毕！")
